src/jvm/main/org/jetbrains/kotlinx/lincheck/runner/ParallelThreadsRunner.kt

```diff
@@ -64,15 +63,31 @@ internal open class ParallelThreadsRunner(
     private fun trySetResumedStatus(iThread: Int, actorId: Int) = completionStatuses[iThread].compareAndSet(actorId, null, CompletionStatus.RESUMED)
     private fun trySetCancelledStatus(iThread: Int, actorId: Int) = completionStatuses[iThread].compareAndSet(actorId, null, CompletionStatus.CANCELLED)
 
-    private val uninitializedThreads = AtomicInteger(scenario.threads) // for threads synchronization
+    private val uninitializedThreads = AtomicInteger(scenario.nThreads) // for threads synchronization
     private var yieldInvokedInOnStart = false
 
+    var currentExecutionPart: ExecutionPart? = null
+        private set
+
+    private lateinit var initialPartExecution: TestThreadExecution
```

Comment:
Do not use `lateinit` in performance-sensitive code

Reply:
Removed it